app/models/asp/payment_request_transition.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module ASP
+  class PaymentRequestTransition < ApplicationRecord
+    include Statesman::Adapters::ActiveRecordTransition
+
+    belongs_to :asp_payment_request, class_name: "ASP::PaymentRequest", inverse_of: :asp_payment_request_transitions
+
+    after_destroy :update_most_recent, if: :most_recent?
+
+    private
+
+    def update_most_recent
+      last_transition = asp_payment_request.asp_payment_request_transitions.order(:sort_key).last
+
+      return if last_transition.blank?
+
+      last_transition.update_column(:most_recent, true) # rubocop:disable Rails/SkipsModelValidations
+    end
```

Comment:
Code répété avec `PfmpTransition`, à factoriser pour rester DRY

Reply:
c'est le [code par défaut de Statesman](https://github.com/gocardless/statesman?tab=readme-ov-file#persistence) je préfère pas trop qu'on y touche for now

Reply:
Je ne propose pas de réinventer le code par défaut de Statesman, juste de factoriser ce qui peut l'être.
Je ne vois pas de bonne raison de ne pas factoriser le code ici.

Reply:
il y a le nom du modèle dans la méthode `update_most_recent` ce qui rend la factorisation un peu délicate. Il faut faire de la metaprog c'est pas le scope de cette PR mais ravi de regarder un autre jour.